### PR TITLE
integration test for pkgdb scrape --rules

### LINKED ISSUE #593: Create an integration test for pkgdb scrape --rules implementation
This test should cover a set of rules like this

```
{
  "allowRecursive": [
    ["legacyPackages", null, "darwin"],
  ],
  "allowPackage": [
    ["legacyPackages", null, "python310Packages", "pip"],
    ["legacyPackages", null, "python310Packages", "django"],
  ],
  "disallowRecursive": [
    ["legacyPackages", null, "python310Packages"],
    ["legacyPackages", null, "emacsPackages"],
    ["legacyPackages", null, "vimPlugins"]
  ],
  "disallowPackage": [
    ["legacyPackages", null, "sqlite"]
  ]
}
```
`allowPackage` takes priority over `disallowRecursive.`

that lets us "disable all python310Packages except for ____"

This ticket contributes toward #547

### PR DESCRIPTION
**Pull request by samrose**
## Proposed Changes

This PR will close #593 

it contributes toward #547 

The test will establish a set of rules serialized in json

```
{
  "allowRecursive": [
    ["legacyPackages", null, "darwin"],
  ],
  "allowPackage": [
    ["legacyPackages", null, "python310Packages", "pip"],
    ["legacyPackages", null, "python310Packages", "django"],
  ],
  "disallowRecursive": [
    ["legacyPackages", null, "python310Packages"],
    ["legacyPackages", null, "emacsPackages"],
    ["legacyPackages", null, "vimPlugins"]
  ],
  "disallowPackage": [
    ["legacyPackages", null, "sqlite"]
  ]
}

```

allowPackage takes priority over disallowRecursive.

that lets us "disable all python310Packages except for ____"

The implementation will make the test pass.

This work will need to be reviewed and adjusted (Possibly in another ticket) against forthcoming design by @ghudgins 

The usage will need to be documented as well, even if it is temporary for the WIP design prior to the design by Graham.

## Release Notes

N/A

### CODE REVIEW

File: pkgdb/CONTRIBUTING.md [hunk 1]
```diff
@@ -10,7 +10,15 @@ pkgdb
...
 ...<SNIP>...
 ```
+## Integration Tests
 
+```shell
+$ nix develop;
+$ cd ./pkgdb;
+$ make -j all;
+$ make bats-check;
+$ command -v flox-pkgdb-tests; #to run the bats tests as an executable
```
Comment (aakropotkin) [inline pkgdb/CONTRIBUTING.md:20]: This command just sanity checks that the test wrapper is in `PATH`.

You can run it by name

**Review by aakropotkin — CHANGES_REQUESTED** [review]
LGTM other than the doc tweak.
Feel free to merge once you have implemented the processor for the rules.

File: pkgdb/CONTRIBUTING.md (hunk 1 above)
Comment (samrose) [reply pkgdb/CONTRIBUTING.md]: should be resolved with https://github.com/flox/flox/pull/594/commits/b054862d380dcdcb178851dbf2f6d1017e36afdc

**Comment by aakropotkin** [comment]
superseded by #714

**Pull request closed without merging**